### PR TITLE
Remove invalid mkv support reported by browserdeviceprofile

### PR DESCRIPTION
**Pull request by YouKnowBlom**

### CODE REVIEW

File: components/maincontroller.js [hunk 1]
```diff
@@ -615,6 +615,22 @@
             }
         );
 
+        //Temporary solution until we can make a custom browserdeviceprofile
```
Comment (cvium) [inline components/maincontroller.js]:
```suggestion
        // Temporary solution until we can make a custom browserdeviceprofile
```

File: components/maincontroller.js [hunk 2]
```diff
@@ -615,6 +615,22 @@
...
+        //webm support is currently mistaken for mkv support
```
Comment (cvium) [inline components/maincontroller.js]:
```suggestion
        // webm support is currently mistaken for mkv support
```

File: components/maincontroller.js [hunk 3]
```diff
@@ -615,6 +615,22 @@
...
+            }
+            return true;
+        });
+
+        profile.TranscodingProfiles = profile.TranscodingProfiles.filter((item) => {            
+            if (item.Container == "mkv") {
+                return false;
+            }
+            return true;
+        })
```
Comment (cvium) [inline components/maincontroller.js]:
```suggestion
        });
```

File: components/maincontroller.js [hunk 4]
```diff
@@ -615,6 +615,22 @@
...
+                return false;
+            }
+            return true;
+        });
+
+        profile.TranscodingProfiles = profile.TranscodingProfiles.filter((item) => {            
+            if (item.Container == "mkv") {
+                return false;
+            }
+            return true;
```
Comment (cvium) [inline components/maincontroller.js]:
```suggestion
            return item.Container !== "mkv";
```

File: components/maincontroller.js [hunk 5]
```diff
@@ -615,6 +615,22 @@
             }
         );
 
+        //Temporary solution until we can make a custom browserdeviceprofile
+        //webm support is currently mistaken for mkv support
+        profile.DirectPlayProfiles = profile.DirectPlayProfiles.filter((item) => {
+            if (item.Container == "mkv") {
+                return false;
+            }
+            return true;
```
Comment (cvium) [inline components/maincontroller.js]:
```suggestion
            return item.Container !== "mkv";
```

File: components/maincontroller.js (hunk 3 above)
Comment (cvium) [inline components/maincontroller.js]:
```suggestion
        profile.TranscodingProfiles = profile.TranscodingProfiles.filter(item => item.Container !== "mkv");
```

File: components/maincontroller.js [hunk 6]
```diff
@@ -615,6 +615,22 @@
...
+        });
```
Comment (cvium) [inline components/maincontroller.js]:
```suggestion
        profile.DirectPlayProfiles = profile.DirectPlayProfiles.filter(item => item.Container !== "mkv");
```

Comment (cvium) [reply components/maincontroller.js]: I am not super strong in ES6, but I think this should work?

**Review by cvium — CHANGES_REQUESTED** [review]

File: components/maincontroller.js (hunk 6 above)
Comment (cvium) [reply components/maincontroller.js]: Could probably also be written as `profile.DirectPlayProfiles = profile.DirectPlayProfiles.filter(({ Container }) => Container !== "mkv");` but I don't think that's any clearer

Comment (YouKnowBlom) [reply components/maincontroller.js]: I do believe that works. I think I just have an old habit of avoiding shorthand things but for this I'd say it's perfect :p

**Comment by YouKnowBlom** [comment]
Should I squash before merge?

**Review by cvium — APPROVED** [review]

**Pull request merged**